src/DGtal/shapes/MeshVoxelizer.h

```diff
@@ -109,15 +109,111 @@ namespace DGtal
     {
       PointR3 myFirst, mySecond;
       Edge() {}
-      Edge(PointR3 mf, PointR3 ms) : myFirst(mf), mySecond(ms) {}
+      Edge(const PointR3& mf, const PointR3& ms) : myFirst(mf), mySecond(ms) {}
+      Edge(const Edge&) = default;
+    };
+
+    /**
+     * Internal Intersection Target structure
+     *
+     * @tparam Dimension dimension of the intersection target
+     * @tparam Separation strategy of the voxelization (6 or 26)
+     * @tparam Dummy dummy parameter just to be able to specialize inside template class (FIXME with another location)
+     */
+    template<size_t Sep, size_t Dimension = 1, typename Dummy = void>
+    struct IntersectionTarget { };
+
+    /**
+     * Specialization for 6-separated with 1D intersection target
+     */
+    template<typename Dummy>
+    struct IntersectionTarget<6, 1, Dummy>
```

Comment:
I think I need to put `IntersectionTarget<>` implementation [(this block)](https://github.com/hadjiszs/DGtal/blob/d3f6b016a6a202ba6b33f83a17389ada01e6d24b/src/DGtal/shapes/MeshVoxelizer.h#L123-L209) out of `MeshVoxelizer<>` class, in a separate `.h/.ih` because here like that it seems not really ok and IntersectionTarget may be grow up in the future if 2d intersection target is implemented

Should I put this code in a separate .ih /.h ? And, in this case, where do you think I should put `"IntersectionTarget".(h|ih)` ? in the same module ? shapes ?